eq-author-api/middleware/identification/verifyJwtToken.js

```diff
@@ -0,0 +1,49 @@
+const fetch = require("node-fetch");
+const jwt = require("jsonwebtoken");
+
+const {
+  VERIFY_PUBLIC_KEYS_URL,
+  VERIFY_VALIDITY_OPTIONS,
+} = require("../../constants/googleServiceKeys");
+
+let googlePublicKeys;
+
+const fetchUpdatedKeys = async () => {
+  try {
+    const response = await fetch(VERIFY_PUBLIC_KEYS_URL);
+
+    googlePublicKeys = await response.json();
+  } catch (error) {
+    //    console.log({ error });
+  }
+};
+fetchUpdatedKeys();
+
+const verifyToken = (accessToken, header, userId) => {
+  const currentPublicKey = googlePublicKeys && googlePublicKeys[header.kid];
```

Comment:
I think you can return `false` early from this function if the `kid` is not in the `Object.keys(googlePublicKeys)`. No need to verify if you detect the key has been rotated.

Reply:
Done